pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -555,77 +552,78 @@ func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMa
 		}
 
 		if expectedContainer == nil {
-			r.logger.Info(fmt.Sprintf("Container '%+v' not found in pod, recreating pod", actualContainer))
-			return true
+			messages = append(messages, []string{fmt.Sprintf("Container '%+v' not found in pod, recreating pod", actualContainer)})
 		}
 
-		if changed := r.compareContainers(*expectedContainer, actualContainer); changed {
-			return true
+		if msg := r.compareContainers(*expectedContainer, actualContainer); len(msg) > 0 && messages != nil {
+			messages[0] = append(messages[0], msg...)
 		}
 	}
 
-	return false
+	return messages
 }
 
-func (r *ReconcileJenkinsBaseConfiguration) compareContainers(expected corev1.Container, actual corev1.Container) bool {
+func (r *ReconcileJenkinsBaseConfiguration) compareContainers(expected corev1.Container, actual corev1.Container) []string {
```

Comment:
should return [][]string with normal and verbose message

Reply:
compareContainers should return always a verbose message imho